feat(runners): Add delay to prevent ssm rate limits using setTimeout

devsh4 opened the pull request:
Adding a fix for bug https://github.com/philips-labs/terraform-aws-github-runner/issues/1841 which prevents hitting AWS SSM rate limits by adding a delay between subsequent putParameter calls.

**Note**: 25ms delay is based on the [AWS service quotas](https://docs.aws.amazon.com/general/latest/gr/ssm.html#limits_ssm) as they only allow 40 requests per second to the parameter store by default.

**[Tests]**:
- Pre commit checks ✅ 
- Ran build locally ✅ 
- Ran linter ✅ 
- Ran jest tests ✅ 

**Related PRs/discussion for reference**:
1. (never got merged): https://github.com/philips-labs/terraform-aws-github-runner/pull/1843
2. (went stale due to inactivity & failing tests):  https://github.com/philips-labs/terraform-aws-github-runner/pull/2305

devsh4 commented:
@npalm @navdeepg2021 Please review this PR so we can merge this fix and unblock ourselves from upgrading the module 🙏

devsh4 commented:
@npalm @GuptaNavdeep1983 Please review this PR so we can merge this fix and unblock everyone out there from upgrading the module 🙏 This is my **third** attempt at merging this.

GuptaNavdeep1983 commented:
> @npalm @GuptaNavdeep1983 Please review this PR so we can merge this fix and unblock everyone out there from upgrading the module 🙏 This is my **third** attempt at merging this.
> 
> In the earlier two attempts, one PR went stale & the test coverage went below threshold. Therefore this PR also adds the required tests.

Thanks for the PR. The PR looks good, Can we also add a condition that checks for the time taken by the test for creation of 40 instances and ensure that the time taken is more than 1000 ms which I did see that is more than 1000 ms? I dont see any other way to test the delay.

devsh4 commented:
@GuptaNavdeep1983 Thanks for the quick review, fixed the formatting and updated the test to check the delay.

**Additional context for posterity**: We have been running the module with this delay since 9 months now to run a pool of 200+ runners daily for our org and it is working seamlessly without any unexpected lambda costs.

GuptaNavdeep1983 commented:
> @GuptaNavdeep1983 Thanks for the quick review, fixed the formatting and updated the test to check the delay. Can you please review again?
> 
> **Additional context for posterity**: We have been running the module with this delay since 9 months now to run a pool of 200+ runners daily for our org in production and it is working seamlessly, without any unexpected lambda costs.

Can you please change the require statement into `import { performance } from 'perf_hooks';`

devsh4 commented:
@GuptaNavdeep1983 Updated, linter is fixed now.

devsh4 commented:
@GuptaNavdeep1983 @npalm can one you'll please approve the workflow? Fixed the linter/formatter and updated the branch to be up-to-date with `main`. Thanks!

devsh4 commented:
@GuptaNavdeep1983 Thanks a lot for reviewing! What's the next step in getting this PR merged?

npalm commented on modules/runners/lambdas/runners/src/aws/runners.test.ts:
```diff
@@ -224,6 +225,75 @@ describe('create runner', () => {
+    await createRunner({ ...createRunnerConfig(defaultRunnerConfig), numberOfRunners: 40 });
+    const endTime = performance.now();
+
+    expect(endTime - startTime).toBeGreaterThan(1000);
```
👍

npalm commented:
changed semantic commit type to feat instead of fix

devsh4 commented:
@npalm Thanks for reviewing! When can I expect the PR to merge? Just curious regarding the process.
